modules/animation/include/modules/animation/datastructures/propertytrack.h

```diff
@@ -146,16 +146,87 @@ void setPropertyFromKeyframeHelper(CameraProperty* property, const CameraKeyfram
 void setKeyframeFromPropertyHelper(const CameraProperty* property, CameraKeyframe* keyframe);
 
 /**
+ * Presses the button.
  * Helper function for inviwo::animation::PropertyTrack::setPropertyFromKeyframe
  * @see inviwo::animation::BasePropertyTrack::setPropertyFromKeyframe
  */
 void setPropertyFromKeyframeHelper(ButtonProperty* property, const ButtonKeyframe* keyframe);
 /**
- * Helper function for inviwo::animation::PropertyTrack::setKeyframeFromProperty
+ * Does nothing. The property of a ButtonKeyframe cannot be changed since all are supposed to be the
+ * same for a ButtonTrack. Helper function for
+ * inviwo::animation::PropertyTrack::setKeyframeFromProperty
  * @see inviwo::animation::BasePropertyTrack::setKeyframeFromProperty
  */
 void setKeyframeFromPropertyHelper(const ButtonProperty* property, ButtonKeyframe* keyframe);
 
+template <typename Prop, typename Seq>
+struct AnimateSequence {
+    /*
+     * Helper function for when we know that we are between keyframes within a KeyframeSequence.
+     * Called from PropertyTrack<Prop, Key, Seq>::operator()(Seconds from, Seconds to,
+     * AnimationState state) const
+     * Provide template specialization of this method if you want custom
+     * property/sequence behaviour.
+     */
+    static AnimationTimeState animate(Prop* prop, const Seq& seq, Seconds from, Seconds to,
+                                      AnimationState state);
+};
+template <typename Prop, typename Seq>
+AnimationTimeState AnimateSequence<Prop, Seq>::animate(Prop* prop, const Seq& seq, Seconds from,
+                                                       Seconds to, AnimationState state) {
+    typename Prop::value_type v;
+    seq(from, to, v);
+    prop->set(v);
+    return {to, state};
+}
+
+template <>
+inline AnimationTimeState
+AnimateSequence<CameraProperty, KeyframeSequenceTyped<CameraKeyframe>>::animate(
```

Comment:
Don't think this belongs in this header file. python in it's own camera track file

Reply:
Note: specialising the member function work as long as it is a full specialisation. i.e. you provide all the template arguments, but it is not supported for partial specialisations then you need to do a partial specialisation of the struct. Which is why I wanted you to break out the functionality into its own struct so one can do partial specialisations. I also prefer that we then also to the specialisation on the class even if it is full to avoid having two different kinds of specialisations around to avoid running into problems when someone tries to do a partial specialisation